src/core/algorithms/fd/hyfd/validator.cpp

```diff
@@ -249,6 +250,30 @@ Validator::FDValidations Validator::ValidateAndExtendSeq(std::vector<LhsPair> co
     return result;
 }
 
+Validator::FDValidations Validator::ValidateAndExtendPar(std::vector<LhsPair> const& vertices) {
+    FDValidations result;
+    boost::asio::thread_pool pool(threads_num_);
+    std::vector<std::future<FDValidations>> validation_futures;
+    validation_futures.reserve(vertices.size());
+
+    for (auto const& vertex : vertices) {
+        std::packaged_task<FDValidations()> task(
+                [this, &vertex]() { return GetValidations(vertex); });
+        validation_futures.push_back(task.get_future());
+        boost::asio::post(pool, std::move(task));
+    }
+
+    pool.join();
+    pool.wait();
```

Comment:
Looking at the documentation, these seem to do the same thing, why is it written this way?